### ISSUE TITLE
New buzzwords This Week in Programming: Big Blues AI Play for the Enterprise

### ISSUE THREAD

**Issue by sudocurse** [sudocurse/reading-list#8971]
This Week in Programming: Big Blue&rsquo;s AI Play for the Enterprise<br>
https://ift.tt/2pBqsDT<br>
<img width="640" height="453" src="https://storage.googleapis.com/cdn.thenewstack.io/media/2018/03/0c6b7a85-merge-sort.png" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2018/03/0c6b7a85-merge-sort.png">
<p>Chances are if you were to ask a budding developer what companies are top of mind for them, one of the world&rsquo;s oldest computing companies may not make the list. Nonetheless, the company that began as the archaic &ldquo;Computing-Tabulating-Recording Company&rdquo; before renaming to the much more modern (sarcasm abounds) &ldquo;International Business Machines Corporation&rdquo; &mdash; you know it as IBM &mdash; is still making headlines with its <a href="https://www.ibm.com/blogs/events/think-2018/day-3-quantum-blockchain-better-future/" class="ext-link" rel="external nofollow">machine learning and quantum computing efforts</a>.</p>
<p>While Big Blue may not be on the tip of your tongue when you think of bleeding edge programming languages, its latest announcements do focus on powering the latest in programming efforts. According to <a href="https://sdtimes.com/ai/ibm-focuses-enterprise-ai-adoption-think-2018/" class="ext-link" rel="external nofollow">SDTimes</a>, the company unveiled this week &ldquo;a number of new solutions designed to help enterprises harness the power of artificial intelligence,&rdquo; including the <a href="https://www.prnewswire.com/news-releases/ibm-launches-watson-data-kits-to-help-accelerate-enterprise-ai-adoption-300616363.html" class="ext-link" rel="external nofollow">IBM Watson Data Kit</a>, and the <a href="https://developer.apple.com/ibm/" class="ext-link" rel="external nofollow">IBM Watson Services for Core ML</a>, which purports to make it &ldquo;easy to build apps that access powerful Watson capabilities right from iPhone and iPad.&rdquo;</p>
<p>According to ZDNet, IBM&rsquo;s real focus here is to <a href="http://www.zdnet.com/article/ibm-think-2018-postmortem-making-incumbent-enterprises-great-again/" class="ext-link" rel="external nofollow">Make Incumbent Enterprises Great Again</a> (groan), also known as &ldquo;catching a ride on the AI money train of big business.&rdquo; Seriously though, at the core of this effort is IBM&rsquo;s <a href="https://www.ibm.com/cloud/watson-studio" class="ext-link" rel="external nofollow">Watson Studio</a>, which ZDNet calls &ldquo;a 2.0 rethink of <a href="https://www.ibm.com/analytics/us/en/watson-data-platform/" class="ext-link" rel="external nofollow">Watson Data Platform</a> and <a href="https://datascience.ibm.com/" class="ext-link" rel="external nofollow">Data Science Experience</a>&rdquo; that &ldquo;provides a more visually integrated user experience&rdquo; while also remaining &ldquo;very much aimed at hard-core data scientists.&rdquo; And for those hard-core data scientists, &ldquo;Watson Studio can automatically generate complex, multi-layered neural networks and automatically run hyperparameter optimization for tuning models that would otherwise require considerable manual footwork.&rdquo; Oh, and support for <a href="https://docs.anaconda.com/anaconda/packages/pkg-docs" class="ext-link" rel="external nofollow">Anaconda packages</a>, to boot.</p>
<p>And for the rest of you who may never really touch an IBM product for years to come, there&rsquo;s plenty of other stuff to talk about this week, so let&rsquo;s get on with it.</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Is your child texting about deep learning?</p>
<p>lol: lots of loss<br>
omg: one more GPU<br>
brb: bringing RNNs back<br>
btw: better try word2vec<br>
lmao: layers &ndash; more and oversized<br>
smdh: save me, Dear Hinton</p>
<p>(inspired by a <a href="https://twitter.com/Iwillleavenow?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">@Iwillleavenow</a> version about infosec)</p>
<p>&mdash; Miles Brundage (@Miles_Brundage) <a href="https://twitter.com/Miles_Brundage/status/975959968196145152?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">March 20, 2018</a></p>
</blockquote>
</div>
<h2>This Week in Programming</h2>
<ul>
<li>
<strong>Java 10 Arrives:</strong> That&rsquo;s right, rejoice (if that&rsquo;s a thing you do over new Java versions) because <a href="https://jaxenter.com/java-10-is-here-142550.html" class="ext-link" rel="external nofollow">Java 10 is here!</a> Though, by the comments on HackerNews, it sure looks like <a href="https://jaxenter.com/java-8-still-strong-java-10-142642.html" class="ext-link" rel="external nofollow">many of you are still using Java 8</a>, so going by that timeline, perhaps you&rsquo;ll get to Java 10 by 2021! Either way, this is the first release following Oracle&rsquo;s newly designed six-month release cycle, (Java 9 was just six months ago) and highlights include local-variable type inference, Parallel Full GC for G1, application Class-Data sharing, and an experimental Java-Based JIT compiler. For full details, make sure to check out Oracle&rsquo;s <a href="https://blogs.oracle.com/java-platform-group/introducing-java-se-10" class="ext-link" rel="external nofollow">introduction to Java SE 10</a>. And for those of you keeping your sights on the horizon, InfoWorld has a piece looking at the <a href="https://www.infoworld.com/article/3265447/java/java-11-roadmap-the-new-features-you-can-expect.html" class="ext-link" rel="external nofollow">Java 11 roadmap</a>.</li>
<li>
<strong>Add Situational Awareness to Android Apps with Google&rsquo;s Transition API:</strong> The idea behind the Activity Recognition Transition API is simple. It lets you know when a user&rsquo;s activity &mdash; walking, running, biking, driving, etc &mdash; has changed. While this might seem like a simple problem, it has a bunch of nuanced variables (as Google notes, should you trust a spike in a non-driving activity or is it a momentary classification error?). Well, Google has a plethora of data on which to base its analyses and now it&rsquo;s <a href="https://android-developers.googleblog.com/2018/03/activity-recognitions-new-transition.html" class="ext-link" rel="external nofollow">available to you as an API</a>. More features will be added in coming months, such as &ldquo;differentiating between road and rail vehicles&rdquo;, but the API is available now. Details are available in the <a href="https://developer.android.com/guide/topics/location/transitions.html" class="ext-link" rel="external nofollow">Transitions API guide</a>.</li>
<li>
<strong>AWS Docs Go To GitHub:</strong> Application Development Trends Magazine reports that Amazon Web Services has put its <a href="https://adtmag.com/articles/2018/03/19/open-source-aws-docs.aspx" class="ext-link" rel="external nofollow">Documentation on GitHub</a> and is now soliciting pull requests &ldquo;as part of an ongoing effort to open source developer guides and other documentation. As part of this overall effort, Amazon had already open sourced <a href="https://aws.amazon.com/tools/" class="ext-link" rel="external nofollow">AWS SDK</a> developer guides in the <a href="https://github.com/awsdocs" class="ext-link" rel="external nofollow">awsdocs</a> repository and has now added more than 138 new developer and user guides. Amazon&rsquo;s blog <a href="https://aws.amazon.com/blogs/aws/aws-documentation-is-now-open-source-and-on-github/" class="ext-link" rel="external nofollow">post</a> last week has the full details.</li>
<li>
<strong>GitLab Brings CI/CD to GitHub:</strong> In what <a href="https://techcrunch.com/2018/03/22/gitlab-adds-support-for-github/" class="ext-link" rel="external nofollow">Techcrunch</a> calls &ldquo;an interesting twist&rdquo;, GitLab 10.6 has been <a href="https://about.gitlab.com/2018/03/22/gitlab-10-6-released/" class="ext-link" rel="external nofollow">released with CI/CD for GitHub and deeper Kubernetes integration</a>. &ldquo;<a href="https://about.gitlab.com/" class="ext-link" rel="external nofollow">GitLab</a>, which in many ways competes with <a href="https://github.com/" class="ext-link" rel="external nofollow">GitHub</a> as a shared code repository service for teams,&rdquo; writes Lardinois, &ldquo;is bringing its continuous integration and delivery (CI/CD) features <a href="https://about.gitlab.com/features/github/" class="ext-link" rel="external nofollow">to GitHub</a>.&rdquo; According to the announcement, continuous integration, delivery and deployment &ldquo;form the backbone of modern DevOps,&rdquo; but &ldquo;one thing that was missing was that you couldn&rsquo;t use GitLab CI/CD with GitHub.&rdquo; If you&rsquo;re interested in trying it out, it will be free to developers until March 22, 2019.</li>
<li>
<strong>LG Open Sources webOS:</strong> Remember <a href="https://en.wikipedia.org/wiki/WebOS" class="ext-link" rel="external nofollow">webOS</a>, that operating system first launched as PalmOS back in 2009? Well, it&rsquo;s been powering the likes of smart fridges and televisions ever since and LG has decided to <a href="https://sdtimes.com/os/lg-launches-open-source-version-webos/" class="ext-link" rel="external nofollow">launch an open source version</a>. According to LG, webOS is &ldquo;now a mature and stable platform ready to move beyond TVs.&rdquo; The <a href="http://webosose.org/discover/webos-ose-overview/" class="ext-link" rel="external nofollow">webOS Open Source Edition is available now</a>.</li>
<li>
<strong>C# 8.0 On The Way:</strong> It looks like C# 8.0 is on the way and developer Kenneth Truyers has taken a look at <a href="https://www.kenneth-truyers.net/2018/03/20/whats-new-c-8-0/" class="ext-link" rel="external nofollow">what&rsquo;s new and coming in C# 8.0</a>. In all, he says that &ldquo;the planned additions will be a significant improvement in making C# more robust and terse&rdquo; including non-nullable reference types by default, the ability to use async/await inside an iterator, the ability to provide an implementation on an interface method, and the ability to create extension properties, fields and operators. The full discussion of the potential features are on <a href="https://github.com/dotnet/csharplang/tree/master/proposals" class="ext-link" rel="external nofollow">GitHub</a>.</li>
</ul>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">In retrospect, it might have been a mistake to give Facebook all of my personal information in exchange for seeing what my high school friends eat for dinner</p>
<p>&mdash; Kelvin Yu (@InternetKelvin) <a href="https://twitter.com/InternetKelvin/status/976092573935714304?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">March 20, 2018</a></p>
</blockquote>
</div>
<ul>
<li>
<strong>Those God Damn Privacy Rules:</strong> (For reference, first <a href="https://twitter.com/NSQE/status/974005089512783872" class="ext-link" rel="external nofollow">see this tweet</a>.) Surely, you&rsquo;ve heard of the GDPR by now, as it goes into full force on May 25th. Well, <a href="https://blog.acolyer.org/2018/03/21/on-purpose-and-by-necessity-compliance-under-the-gdpr/" class="ext-link" rel="external nofollow">Adrian Coyler takes a look at GDPR compliance</a> this week in his daily review of academic papers, though he doesn&rsquo;t seem to buy the whole premise sold by most on how to prepare for GDPR. &ldquo;The central idea in the paper is that business process modeling can be a powerful tool for managing purpose and compliance. I&rsquo;m not sure how practical that is when your architecture looks like a death star and you&rsquo;re on a journey towards 10,000 deploys a day, but you&rsquo;re certainly going to need some way of tracking and managing data, consent, and purpose,&rdquo; Coyler writes. &ldquo;I take a somewhat different viewpoint here: for many organizations, I contend that an approach requiring a complete, accurate, up-to-date, global understanding of all processes and data flows is probably doomed to failure, as this is an impossible task! And yet we can&rsquo;t escape the challenges the authors (and the regulations) place before us. My personal sense is that we need to track the provenance of data as it flows through an organization, including proof of consent and the purposes for which consent was given, and then match the purpose of a process against that consent record.&rdquo;</li>
<li>
<strong>Are APIs a Solution to GDPR?</strong> Looking further at the same topic, perhaps from a slightly different angle, API Evangelist Kin Lane offers up that the <a href="https://apievangelist.com/2018/03/19/gdpr-forcing-us-to-ask-questions-about-our-data/" class="ext-link" rel="external nofollow">GDPR is forcing us to ask questions about our data</a>. &ldquo;Most companies collect huge amounts of data, believing it is essential to the value they bring to the table, without no real understanding of everything that is being collected, and any logical reasons behind why it is gathered, stored, and kept around,&rdquo; he writes. However, &ldquo;doing APIs and becoming GDPR compliant go hand in hand. To do APIs you need to map out the data landscape across your organization, something that will contribute to GDPR. To respond to GDPR events, you will need APIs that provide access to end-users data, and leverage API authentication protocols like OAuth to ensure partnerships, and 3rd party access to end-users data are accountable.&rdquo;</li>
</ul>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Tech buzzwords explained:<br>
AI&mdash;regression<br>
Big data&mdash;data<br>
Blockchain&mdash;database<br>
Algorithm&mdash;automated decision-making<br>
Cloud&mdash;Internet<br>
Crypto&mdash;cryptocurrency<br>
Dark web&mdash;Onion service<br>
Data science&mdash;statistics done by nonstatisticians<br>
Disruption&mdash;competition<br>
Viral&mdash;popular<br>
IoT&mdash;malware-ready device</p>
<p>&mdash; Arvind Narayanan (@random_walker) <a href="https://twitter.com/random_walker/status/976836626121977858?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">March 22, 2018</a></p>
</blockquote>
</div>
<ul>
<li>
<strong>IKEA Instructions for Algorithms:</strong> Finally, let&rsquo;s finish up this week with a few fun tidbits. First up, <a href="https://idea-instructions.com/" class="ext-link" rel="external nofollow">IDEA instructions</a>, &ldquo;an ongoing series of nonverbal algorithm assembly instructions&rdquo; that bite off the IKEA style of design. That&rsquo;s right, if you&rsquo;ve ever wanted to know how a Merge Sort algorithm works &mdash; without all those troublesome words and numbers, this is your go-to spot.</li>
<li>
<strong>Exercises in Programming Style:</strong> If you combine programming and literature, you get this literary movement out of the 1960s called Oulipo. In essence, it was like algorithmically created writing, at times. It was writing created through limitation and formula. And one such product of this form was created by one of Oulipo&rsquo;s founding members, Raymond Queneau, <a href="https://en.wikipedia.org/wiki/Exercises_in_Style" class="ext-link" rel="external nofollow">Exercises in Style</a>. In it, the same story is told over and over again, but with slight variations on either the point of view or the type of language used, demonstrating how the language used formed the central truth &mdash; if there were a central truth to be found. Well, now there&rsquo;s an <a href="https://dzone.com/articles/exercises-in-programming-style" class="ext-link" rel="external nofollow">Exercises in Programming Style</a>, which consists of a simple program implemented in 33 different programming styles. I don&rsquo;t know about you, but that just appeals to my brain.</li>
</ul>
<p class="attribution">Feature image from <a href="https://idea-instructions.com/" class="ext-link" rel="external nofollow">IDEA</a>, on <a href="https://idea-instructions.com/merge-sort/" class="ext-link" rel="external nofollow">Merge Sort</a> algorithm.</p>
<p>The post <a rel="nofollow" href="https://thenewstack.io/this-week-in-programming-big-blues-ai-play-for-the-enterprise/">This Week in Programming: Big Blue&rsquo;s AI Play for the Enterprise</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br>
<br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
March 24, 2018 at 09:00AM
